### ISSUE TITLE
Basic auth working locally but not on the server

### ISSUE THREAD

**Issue by lnjuanj** [mevdschee/php-crud-api#552]
Hi,
I have been looking for some days for the solution to this one: I have php.api working fine with basicAuth middleware on my local machine.

When I upload it to my Plesk server, then I cannot make it work. More specifically, the response to my `"Authorization: Basic"` sends me the `Realm` header, `WWW-Authenticate →Basic realm="Username and password required"`
and I get a *401 response* with the body message: `{"code":1011,"message":"Authentication required"}`
The server also sends me a cookie (FYI).

This is what I am trying to achieve: when the user logs in in a website, I (the API) send the auth header for her/him. The user gets a cookie as a response and the user is now able to send the rest of calls without sending any auth header till the session expires.
This is working like a charm on my computer.

But, on the plesk 17.0.17 server with Ubuntu 16.04.6 LTS‬ (same api.php file as on my machine, apache2 as on my machine, php-fpm 7.1), the answer to the initial request is: 401 Unauthorized, etc. as posted at the beginning of this (please please) "help request" issue.

If I, instead of sending a POST request as I was doing, go to the api.php website, I find the basic auth prompt. No matter what I type, it will reopen again.

Any ideas on how I could find the origin of my problem?
Thanks in advance!

**Comment by mevdschee** [comment]
It is clear that the basic auth is not working. This could be caused by a (plesk/fpm/apache/php) config setting. Maybe try the simple basic auth from this SO post:

https://stackoverflow.com/questions/4150507/how-can-i-use-basic-http-authentication-in-php

Please let me know whether or not such a simple basic auth is working on the server.

**Comment by mevdschee** [comment]
Closing due to inactivity.

**Comment by GerbenWartena** [comment]
For documentation purposes: I had the same issue when trying to use 'basic authentication'. Apparently the Apache server of my web hosting provider stripped the 'Authorization' header from the request.
First I used a 'X-Authorization' header, but then I needed to change the code of the CRUD-API.
Secondly I found a solution changing the .htaccess file: https://stackoverflow.com/questions/26475885/authorization-header-missing-in-php-post-request.
